lib/screens/v2/widget_instance/alert.ex

```diff
@@ -168,8 +168,16 @@ defmodule Screens.V2.WidgetInstance.Alert do
     end
   end
 
-  def valid_candidate?(t) do
-    priority(t) != :no_render
+  def valid_candidate?(%__MODULE__{screen: %Screen{app_id: screen_type}} = t) do
+    cond do
+      screen_type in [:bus_shelter_v2, :bus_eink_v2] ->
+        priority(t) != :no_render and
+          location(t) == :inside and
+          active?(t)
+
+      screen_type == :gl_eink_v2 ->
+        priority(t) != :no_render
+    end
   end
```

Comment:
This works! Just as an exercise though, see if you can split the behavior into two separate function clauses based on the `app_id` value. Hint: you can check if a variable from the function args is in a list of allowed values using a guard like so
```ex
def f(%{key: value} = t) when value in [:some, "list", true] do ...
```

Often if you find yourself checking equality in an `if` or `cond` expression, it could likely be replaced by pattern matching/guards in a `case` expression or the function head.